src/openforms/submissions/api/validation.py

```diff
@@ -34,6 +34,8 @@ class CompletionValidationSerializer(serializers.Serializer):
     privacy_policy_accepted = serializers.BooleanField()
     contains_blocked_steps = serializers.BooleanField()
 
+    # TODO: Move validators to view?
+
```

Comment:
:thinking:  SubmissionAllowed is very much like the SuspensionAllowed. But accepting the privacy policy isn't. @sergei-maertens ?

Reply:
yeah we discussed this yesterday - cleaning this up right is out of scope (and the submit/complete serializer is still marked experimental so we can get away with it), but the privacy policy is something the client can fix/control, hence an HTTP 400 is appropriate, while the submission being blocked is more of a permission thing they have little control over, hence an HTTP 403 would be more appropriate.

HOWEVER - the steps being blocked CAN be derived from user input and logic, so a 400 may be appropriate if/when giving different answers unblocks submission. So it's a bit of a grey area :grimacing: Probably this should be broken up with the _complete endpoint checking if its outright blocked on the form -> HTTP 403 and otherwise if it's dynamically blocked -> HTTP 400 and leave it to the serializer.